lib/jekyll/strapi/collection.rb

```diff
@@ -20,27 +20,35 @@ def generate?
       def each
         # Initialize the HTTP query
         path = "/#{@config['type'] || @collection_name}?_limit=10000"
-        uri = URI("#{@site.endpoint}#{path}")
-        Jekyll.logger.info "Jekyll Strapi:", "Fetching entries from #{uri}"
+        uri = URI("#{@site.endpoint}/api#{path}")
         # Get entries
-        response = Net::HTTP.get_response(uri)
+        Jekyll.logger.info "Jekyll Strapi:", "Fetching entries from #{uri}"
+        # response = Net::HTTP.get_response(uri)
+        response = strapi_request(uri)
         # Check response code
         if response.code == "200"
-          result = JSON.parse(response.body, object_class: OpenStruct)
+          _result = JSON.parse(response.body, object_class: OpenStruct)
+          result = _result.data 
         elsif response.code == "401"
           raise "The Strapi server sent a error with the following status: #{response.code}. Please make sure you authorized the API access in the Users & Permissions section of the Strapi admin panel."
         else
           raise "The Strapi server sent a error with the following status: #{response.code}. Please make sure it is correctly running."
         end
-
         # Add necessary properties
         result.each do |document|
           document.type = collection_name
           document.collection = collection_name
           document.id ||= document._id
+          document.title = document.attributes.Title
+          # During the iteration we pull the whole document using populate=*
+          path_document =         path = "/#{@collection_name}/#{document.id}"
+          uri_document = URI("#{@site.endpoint}/api/#{collection_name}/#{document.id}?populate=*")
```

Comment:
Why do you populate all data in the codebase? I can still retrieve posts without having populate=*

I use: 

```
path = "/#{@config['type'] || @collection_name}?_limit=10000"
```

Reply:
I am not sure how you do this - in docs it says for medias and relationships populate is needed: https://docs.strapi.io/developer-docs/latest/developer-resources/database-apis-reference/rest/populating-fields.html#population 

Also, I am not sure why you are using "_limit", since now I think there is sifferent parameter for pagination. 

I will be touching pagination this weekend to fix it up, I haven't dig it up yet - so I am may be wrong here.